mlflow/xgboost.py

```diff
@@ -130,7 +138,22 @@ def save_model(
                           model. The given example will be converted to a Pandas DataFrame and then
                           serialized to json using the Pandas split-oriented format. Bytes are
                           base64-encoded.
+    :param pip_requirements: Overwrites pip requirements of the conda environment this model should
+        be run in. Accepts either a string path to a pip requirements file on the local filesystem
+        or an iterable of pip requirement strings (e.g. :code:`["pandas", "-r requirements.txt"]`).
```

Comment:
If a user specifies `-r requirements.txt`, they'll need to bundle `requirements.txt` with the model manually (and presumably use an absolute path). Perhaps we should use an example that will work out of the box, since this one is tricky to get right. Can we replace this with a pinned library installation, such as `xgboost==1.4.2`?

Reply:
They don't need to. The currently approach extract packages from a requirements file. For example,

```
# Mix of both shell and python

>>> cat my_requirements.txt
xgboost
scikit-learn

>>> save_model(pip_requiments=["-r my_requirements.txt"])
```

The code above logs a requirements file that looks like:

```
xgboost
scikit-learn
```

Reply:
Alternative approach is detect requirements files and bundle them with the model.

Reply:
Got it - apologies, misread the logic a bit. Can we add some inline comments for clarity?